rust/lance/src/dataset.rs

```diff
@@ -1065,6 +1067,40 @@ impl Dataset {
         }
     }
 
+    /// Get a stream of batches based on iterator of slices of row numbers.
```

Comment:
I read "iterator of slices of row numbers" as `Iterator<Item = &[u64]>`.  I think you are taking in an iterator or row number ranges.

Reply:
Rewriting as range.